### PR TITLE
fix: used `suggestedDecimals` in all possible places.

### PR DESCRIPTION
**Pull request by zhyd1997**
## Related issue

Resolves: #1682

### CODE REVIEW

**Comment by platschi** [comment]
Thanks! I suggest we adjust `suggestedDecimals` to something like this:
```ts
export const suggestedDecimals = (value: WeiSource) => {
	value = wei(value).toNumber();
	if (value >= 10000) return 0;
	if (value >= 10) return 2;
	if (value >= 0.001) return 4;
	return 5;
};
```

@avclarke what do you think? Any implications with this changes for the Exchange?

With the current solution, however, `$0.00` placeholders are now set to 5 decimals by default all over the place. 

Screenshot 2022-11-29 at 11 43 56

We should catch this, maybe by adding something like `	if (value === 0) return 2;` this way dust amounts should still show with up to 5 decimals on the Exchange.

Screenshot 2022-11-29 at 11 48 36

**Comment by zhyd1997** [comment]
@platschi 👍🏿

**Comment by platschi** [comment]
Update suggestion from @Marko-roy :

```ts
export const suggestedDecimals = (value: WeiSource) => {
    value = wei(value).toNumber();
    if (value >= 10000) return 0;
    if (value >= 4) return 2;
    if (value >= 0.1) return 4;
    if (value >= 0.01) return 5;
    return 6;
};
```

**Review by platschi — APPROVED** [review]
Looks good on UI

File: utils/formatters/number.ts [hunk 1]
```diff
@@ -215,10 +209,10 @@ export const weiFromWei = (weiAmount: WeiSource) => {
 export const suggestedDecimals = (value: WeiSource) => {
 	value = wei(value).toNumber();
 	if (value >= 10000) return 0;
-	if (value >= 1) return 2;
-	if (value >= 0.01) return 3;
-	if (value >= 0.001) return 4;
-	return 5;
+	if (value >= 4 || value === 0) return 2;
```
Comment (avclarke) [inline utils/formatters/number.ts:212]: Bigger than 4 seems like a strange rule, is this mean to be 10 or something?

Comment (zhyd1997) [reply utils/formatters/number.ts:212]: Although seems strange, that's correct.

Comment (platschi) [reply utils/formatters/number.ts:212]: Yeah it's correct, I double checked with Marko.

Comment (avclarke) [reply utils/formatters/number.ts:212]: I'll ask Marko what his reasoning his

**Review by platschi — APPROVED** [review]

**Pull request merged**